### PR TITLE
add config option for food voucher

### PR DESCRIPTION
**Pull request by xuwhite**
more config options :tada:

### CODE REVIEW

File: includes/controller/users_controller.php [hunk 1]
```diff
@@ -33,7 +33,7 @@ function users_controller()
         case 'delete':
             return user_delete_controller();
         case 'edit_vouchers':
-            return user_edit_vouchers_controller();
+            return user_edit_vouchers_controller() ;
```
Comment (weeman1337) [inline includes/controller/users_controller.php:36]: Xuff
```suggestion
            return user_edit_vouchers_controller();
```

File: src/Middleware/LegacyMiddleware.php [hunk 2]
```diff
@@ -83,7 +83,7 @@ public function process(
             $content = $translator->translate('page.404.text');
         }
 
-        return $this->renderPage($page, $title, $content);
+        return  $this->renderPage($page, $title, $content);
```
Comment (weeman1337) [inline src/Middleware/LegacyMiddleware.php:86]: Xuff
```suggestion
        return $this->renderPage($page, $title, $content);
```

File: src/Middleware/LegacyMiddleware.php [hunk 3]
```diff
@@ -101,7 +101,7 @@ protected function loadPage($page)
                 user_ical();
                 break;
             case 'atom':
-                require_once realpath(__DIR__ . '/../../includes/pages/user_atom.php');
+                require_once realpath(__DIR__ . '/../../includes/pages/user_stringatom.php');
```
Comment (weeman1337) [inline src/Middleware/LegacyMiddleware.php:104]: Xuff
```suggestion
                require_once realpath(__DIR__ . '/../../includes/pages/user_atom.php');
```

**Review by weeman1337 — CHANGES_REQUESTED** [review]

File: includes/view/User_view.php [hunk 4]
```diff
@@ -166,23 +166,25 @@ function User_delete_view($user)
...
-        ), true),
-        form(
-            [
-                form_spinner('vouchers', __('Number of vouchers given out'), $user->state->got_voucher),
-                form_submit('submit', __('Save'))
-            ],
-            page_link_to('users', ['action' => 'edit_vouchers', 'user_id' => $user->id])
-        )
-    ]);
+    if (config('enable_vouchers')) {
```
Comment (MyIgel) [inline includes/view/User_view.php:169]: A direct return would be better here:
```
if (!config('enable_vouchers')) {
    return '';
}
```

File: includes/view/User_view.php [hunk 5]
```diff
@@ -641,13 +643,17 @@ function User_view(
...
                                 form_submit('submit', __('arrived'), '', false, 'primary')
                             ], page_link_to('admin_arrive'), true) : '',
-                        $admin_user_privilege ? button(
-                            page_link_to(
-                                'users',
-                                ['action' => 'edit_vouchers', 'user_id' => $user_source->id]
-                            ),
-                            icon('file-binary-fill') . __('Edit vouchers')
-                        ) : '',
+                        $admin_user_privilege ?
```
Comment (MyIgel) [inline includes/view/User_view.php:646]: Using ` $admin_user_privilege && config('enable_voucher') ? [...]` would be easier :)

**Review by MyIgel — CHANGES_REQUESTED** [review]

File: includes/view/User_view.php (hunk 4 above)
Comment (xuwhite) [reply includes/view/User_view.php]: done

File: includes/view/User_view.php (hunk 5 above)
Comment (xuwhite) [reply includes/view/User_view.php]: done

File: includes/view/User_view.php [hunk 6]
```diff
@@ -166,6 +166,9 @@ function User_delete_view($user)
  */
 function User_edit_vouchers_view($user)
 {
+    if (!config('enable_vouchers')) {
```
Comment (MyIgel) [inline includes/view/User_view.php:169]:
```suggestion
    if (!config('enable_voucher')) {
```

**Pull request merged**